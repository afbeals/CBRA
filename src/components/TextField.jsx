// Extenal
import React from 'react';
import PropTypes from 'prop-types';
import MaterialTextField from '@material-ui/core/TextField';

// Constants
const classname = 'textField';

// Component
const TextField = ({ leadIcon, className, hasError, ...props }) => (
  <MaterialTextField
    {...props}
    className={`${className ? `${className} ` : ''}${classname}`}
    color="primary"
    error={hasError}
    margin="dense"
    variant="filled"
    data-testid="textfield"
  />
);

TextField.propTypes = {
  className: PropTypes.string,
  hasError: PropTypes.bool,
  leadIcon: PropTypes.node,
};

TextField.defaultProps = {
  hasError: false,
};

export default TextField;

export { TextField };
