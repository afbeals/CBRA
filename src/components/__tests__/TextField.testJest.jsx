// External
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

// Local
import TextField from '../TextField';

// tests
describe('TextField', () => {
  afterEach(cleanup);
  it('should match TextField to snapshot', () => {
    const { getByTestId } = render(<TextField />);
    const textField = getByTestId('textfield');
    expect(textField).toMatchSnapshot();
  });

  it('should prepend the provided classname', () => {
    const { getByTestId } = render(<TextField className="jestTest" />);
    const textField = getByTestId('textfield');
    expect(textField.classList.contains('jestTest')).toEqual(true);
  });

  it('should match Filter to snapshot', () => {
    const { getByTestId } = render(<TextField className={'parent'} />);
    const textfield = getByTestId('textfield');
    const input = textfield.querySelector('input');
    fireEvent.change(input, { target: { value: 'Bul' } });
    expect(input.value).toBe('Bul');
  });
});
