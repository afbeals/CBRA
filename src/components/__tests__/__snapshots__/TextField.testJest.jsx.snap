// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should match TextField to snapshot 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root textField MuiFormControl-marginDense"
  data-testid="textfield"
>
  <div
    class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
      type="text"
      value=""
    />
  </div>
</div>
`;
